File: Capstone_Restaurant_Orders_Portal/backend/src/main/java/com/example/restaurant_order_portal/service/CategoryService.java
package com.example.restaurant_order_portal.service;

import com.example.restaurant_order_portal.dto.CategoryRequestDTO;
import com.example.restaurant_order_portal.dto.CategoryResponseDTO;
import com.example.restaurant_order_portal.entity.Category;

import java.util.List;

/**
 * Service interface for Category operations.
 *
 * Defines business logic methods for managing categories
 * under a specific restaurant.
 */
public interface CategoryService {
    /**
     * Create a new category under a restaurant
     */
    CategoryResponseDTO createCategory(CategoryRequestDTO categoryRequestDTO);

    /**
     * Get all categories belonging to a specific restaurant
     */
    List<CategoryResponseDTO> getCategoriesByRestaurant(Long restaurantId);

    /**
     * Update an existing category
     */
    CategoryResponseDTO updateCategory(Long id, CategoryRequestDTO categoryRequestDTO);

    /**
     * Delete a category by ID
     */
    void deleteCategory(Long id);
}

File: Capstone_Assessment_Portal/frontend/src/services/categoryService.js
/**
 * Category Service
 * Handles all API requests related to category management
 */

import api from "./api";
import { API_ENDPOINTS } from "../constants/apiEndpoints";

/**
 * Fetch all categories
 */
export const getCategories = async () => {
    const response = await api.get(`${API_ENDPOINTS.CATEGORIES}/`);
    return response.data;
};

/**
 * Create a new category
 */
export const createCategory = async (categoryData) => {
    const response = await api.post(
        `${API_ENDPOINTS.CATEGORIES}/`,
        categoryData,
    );

    return response.data;
};

/**
 * Update an existing category
 */
export const updateCategory = async (
    categoryId,
    categoryData,
) => {
    const response = await api.put(
        `${API_ENDPOINTS.CATEGORIES}/${categoryId}`,
        categoryData,
    );

    return response.data;
};

/**
 * Delete category
 */
export const deleteCategory = async (categoryId) => {
    const response = await api.delete(
        `${API_ENDPOINTS.CATEGORIES}/${categoryId}`,
    );

    return response.data;
};